### PR TITLE
feat: implemented DemandAndCapacityNotification frontend

### LINKED ISSUE #315: Implement Message Frontend
**As** PURIS user ,
**I want** a view for Messages ,
**so that** I can get an overview over incoming and outgoing messages .
**Is blocked by:** #316 

## Hints / Details

- add notifications as a new menu entry "Notifications"
- notifications should be displayed as 2 tabs (incoming and outgoing)
- each tab displays the notifications in a table
- outgoing has a button to create new notifications

### Outcome

- Messages are separated into incoming and outgoing
- Messages can be sorted by individual properties
- by default newest messages are at the top

### Acceptance Criteria

- TBD

### PR DESCRIPTION
**Pull request by ReneSchroederLJ**
## Description

- implemented a new view for displaying DemandAndCapacityNotifications
- notifications are divided in incoming and outgoing notifications
- new notifications can be created through a modal

resolves #315

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: frontend/src/features/notifications/components/NotificationInformationModal.tsx [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+    const { partners } = useAllPartners();
+    const { partnerMaterials } = usePartnerMaterials(temporaryDemandCapacityNotification.partnerBpnl ?? `BPNL`);
+
+    const [notifications, setNotifications] = useState<Notification[]>([]);
+    const [formError, setFormError] = useState(false);
+
+    const { sites } = useSites();
+
+    useEffect(() => {
+        setTemporaryDemandCapacityNotification(prevState => ({
```
Comment (tom-rm-meyer-ISST) [inline frontend/src/features/notifications/components/NotificationInformationModal.tsx:116]: Does this make sense? 
- Create notification > save
- Create notification > same notification is defaulted > save -> no update but creation of a new

At least defaulting looks a little bit strange + doesn't work when selecting a notification and then clicking on send

File: frontend/src/views/DemandCapacityNotificationView.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                <h1 className="text-3xl font-semibold text-gray-700 mb-10">Demand And Capacity Notifications</h1>
+                <Stack width='100%' direction="row" justifyContent="space-between">
+                    <Tabs value={selectedTab} onChange={(_, value: number) => setSelectedTab(value)}>
+                        {tabs.map((tab, index) => <Tab key={index} label={tab} />)}
+                    </Tabs>
+                    <Button variant="contained" onClick={() => {
+                        setSelectedNotification(null);
+                        setModalOpen(true)
+                    }}>
+                        <Add></Add> Send Notification
```
Comment (tom-rm-meyer-ISST) [inline frontend/src/views/DemandCapacityNotificationView.tsx:77]: No must but interested into your opinion: Does it make sense to use a paperplane icon or similar as it's "Send" not add? Both is reasonable

File: frontend/src/views/DemandCapacityNotificationView.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        <Box width="100%">
+            <Table
+                onRowClick={(value) => {
+                    onRowSelected(value.row);
+                }}
+                noRowsMsg='No Notifications found'
+                title="Demand and Capacity Notifications"
+                columns={[
+                    { headerName: 'Text', field: 'text', width: 200 },
+                    { headerName: 'Partners Bpnl', field: 'partnerBpnl', width: 200 },
```
Comment (tom-rm-meyer-ISST) [inline frontend/src/views/DemandCapacityNotificationView.tsx:118]: Partner's BPNL or  Partner BPNL

File: frontend/src/features/notifications/components/NotificationInformationModal.tsx [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+import { postDemandAndCapacityNotification } from '@services/demand-capacity-notification';
+import { Notification } from '@models/types/data/notification';
+import { EFFECTS } from '@models/constants/effects';
+import { useAllPartners } from '@hooks/useAllPartners';
+import { LEADING_ROOT_CAUSE } from '@models/constants/leading-root-causes';
+import { STATUS } from '@models/constants/status';
+import { DemandCapacityNotification } from '@models/types/data/demand-capacity-notification';
+import { Site } from '@models/types/edc/site';
+import { useSites } from '@features/stock-view/hooks/useSites';
+import { usePartnerMaterials } from '@hooks/usePartnerMaterials';
```
Comment (tom-rm-meyer-ISST) [inline frontend/src/features/notifications/components/NotificationInformationModal.tsx:35]: As a material can have both directions, do we then send the message twice? The reselling is pretty strange in that case. Maybe we don't need to care about.

**Review by tom-rm-meyer-ISST — CHANGES_REQUESTED** [review]
Thanks for contributing. Great Quality! Just had a few findings or questions. please check

File: frontend/src/features/notifications/components/NotificationInformationModal.tsx (hunk 4 above)
Comment (ReneSchroederLJ) [reply frontend/src/features/notifications/components/NotificationInformationModal.tsx:35]: The way I see it, direction is implied by `effect`. Sending a message with the same effect in both directions makes little sense and if the effect is different then it needs to be 2 notifications in any case.

I'm gladly up for discussing this topic if this logic isn't sufficient.

Comment (tom-rm-meyer-ISST) [reply frontend/src/features/notifications/components/NotificationInformationModal.tsx:35]: Ah fair, didn't had that in mind. Thanks for explaining!

**Review by tom-rm-meyer-ISST — APPROVED** [review]
LGTM, thanks for your efforts!

**Pull request merged**